### PR TITLE
Convert ha-url-sync to TS

### PR DESCRIPTION
**Pull request by balloob**

### CODE REVIEW

File: src/util/ha-url-sync.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        console.log("deselect entity");
+      }
+      this._ignoreNextHassChange = true;
+      fireEvent(this, "hass-more-info", { entityId: null });
+    }
+  };
+}
+
+if (!__DEMO__) {
+  customElements.define("ha-url-sync", HaUrlSync);
```
Comment (iantrich) [inline src/util/ha-url-sync.ts]: use decorator

File: src/util/ha-url-sync.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+/* tslint:disable:no-console */
+const DEBUG = false;
+
+class HaUrlSync extends HTMLElement {
+  private _hass?: HomeAssistant;
+  private _ignoreNextHassChange = false;
+  private _moreInfoOpenedFromPath?: string;
+  private _ignoreNextPopstate = false;
+
+  public connectedCallback() {
```
Comment (iantrich) [inline src/util/ha-url-sync.ts]:
```suggestion
  public connectedCallback(): void {
```

File: src/util/ha-url-sync.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    window.addEventListener("popstate", this._popstateChangeListener);
+  }
+
+  public disconnectedCallback() {
```
Comment (iantrich) [inline src/util/ha-url-sync.ts]:
```suggestion
  public disconnectedCallback(): void {
```

File: src/util/ha-url-sync.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    window.removeEventListener("popstate", this._popstateChangeListener);
+  }
+
+  set hass(newHass) {
```
Comment (iantrich) [inline src/util/ha-url-sync.ts]:
```suggestion
  set hass(newHass: HomeAssistant) {
```

**Review by iantrich — APPROVED** [review]
👍 after comments addressed

File: src/util/ha-url-sync.ts (hunk 1 above)
Comment (balloob) [reply src/util/ha-url-sync.ts]: Can't, it's conditional.

**Pull request merged**